### ISSUE TITLE
Get store in network-interfaces/default.js

### ISSUE THREAD

**Issue by frankspin89** [nuxt-community/apollo-module#6]
I have the following network-interfaces/default.js file

```
import { createNetworkInterface } from 'apollo-client'
import { SubscriptionClient, addGraphQLSubscriptions } from 'subscriptions-transport-ws'

const ssr = process.server

// Create the network interface
const networkInterface = createNetworkInterface({
  uri: '',
  transportBatching: true,
});

networkInterface.use([{
  applyMiddleware(req, next) {
    if (!req.options.headers) {
      req.options.headers = {};  // Create the header object if needed.
    }
    
    const token = // need the store here. 


    req.options.headers.authorization = token ? `Bearer ${token}` : null;
    next();
  }
}]);

let networkInterfaceWithSubscriptions;

// Create the subscription websocket client
if (! ssr) {
  const wsClient = new SubscriptionClient('', {
    reconnect: true,
    timeout: 20000
  });

  // Extend the network interface with the subscription client
  networkInterfaceWithSubscriptions = addGraphQLSubscriptions(
    networkInterface,
    wsClient,
  );
}

const networkInterfaceObj = ssr ? networkInterface : networkInterfaceWithSubscriptions

export default networkInterfaceObj
```

How can I call the vuex store context here? So I can set the Bearer  token from the vuex store

I use the nuxtServerInit function to set the token to the store

```
   nuxtServerInit({ commit }, { req }) {
      if (req.session && req.session.token) {
         commit('SET_TOKEN', req.session.token)
      }
    }
```

**Comment by Samuell1** [comment]
I have same issue, i founded temp workaround to get it from localStorage

**Comment by frankspin89** [comment]
The problem is that localStorage is not available on the server side. So only client-side queries are authenticated with the Bearer token.

**Comment by Samuell1** [comment]
@frankspin89 yes, but i think now is not other solution

**Comment by dohomi** [comment]
@frankspin89 I did it in following way: https://github.com/nuxt-community/apollo-module/issues/5#issuecomment-322994904

**Comment by Samuell1** [comment]
I founded solution https://github.com/Samuell1/apollo-module/tree/get-store-in-networkinterface

then simply in network interface get params:
```js
export default (store) => {
  const networkInterface = createNetworkInterface({
    uri: 'http://192.168.11.240:4000/graphql'
  })

  const authorization = {
    applyMiddleware (req, next) {
      const token = store.state.token
      if (!req.options.headers) {
        req.options.headers = {}
      }
      if (token) {
        req.options.headers.authorization = `Bearer ${token}`
      }
      next()
    }
  }
  networkInterface.use([authorization])
}
```

**Comment by dohomi** [comment]
I added a PR which will change the behaviour like @Samuell1 suggesting in his PR: https://github.com/nuxt-community/apollo-module/pull/18

**Comment by dohomi** [comment]
I think this issue is resolved in v2.0
